druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/dynamic/schema/DynamicLookupProtobufSchemaSerDe.java

```diff
@@ -1,31 +1,81 @@
 package com.yahoo.maha.maha_druid_lookups.query.lookup.dynamic.schema;
 
 import com.fasterxml.jackson.databind.JsonNode;
+import com.google.protobuf.DescriptorProtos;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.yahoo.maha.maha_druid_lookups.query.lookup.namespace.RocksDBExtractionNamespace;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.druid.java.util.common.logger.Logger;
 
+import java.io.FileInputStream;
+import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
 
-public class DynamicLookupProtobufSchemaSerDe implements DynamicLookupCoreSchema {
+public class DynamicLookupProtobufSchemaSerDe implements DynamicLookupCoreSchema<String,Optional<Long>> {
+    private static final Logger LOG = new Logger(DynamicLookupProtobufSchemaSerDe.class);
 
-    private JsonNode coreSchema;
-    //WIP
-    public DynamicLookupProtobufSchemaSerDe(JsonNode coreSchema){
+    private final JsonNode coreSchema;
+    private Map<String,String> schemaPathAsMap ;
+    private Descriptors.Descriptor protobufMessageDescriptor;
+
+    public DynamicLookupProtobufSchemaSerDe(JsonNode coreSchema) throws IOException, Descriptors.DescriptorValidationException {
         this.coreSchema = coreSchema;
+        schemaPathAsMap =  new HashMap<String, String>();
+        DescriptorProtos.FileDescriptorSet set = null;
+        String path ="";
+
+        try {
+            path = getSchema();
+            FileInputStream fin = new FileInputStream(path);
```

Comment:
Do you need to close the stream or is it done by another function below?

Reply:
fixed